portal/src/main/webapp/js/api/cbioportal-datamanager.js

```diff
@@ -1040,7 +1041,16 @@ window.initDatamanager = function (genetic_profile_ids, oql_query, cancer_study_
 	    return this.oql_query;
 	},
 	'getQueryGenes': function () {
-	    return OQL.genes(this.oql_query);
+		if (this.oql_query.trim().length > 0) {
+		    return OQL.genes(this.oql_query);
+		} 
+		return null;
+	},
+	'getQueryGenesets': function () {
+		if (this.geneset_ids.trim().length > 0) {
```

Comment:
Similarly to `this.oql_query`, `geneset_ids` comes also from trimming another string ([line 180 from `global_variables.js`](https://github.com/thehyve/cbioportal/blob/geneset_frontend/portal/src/main/webapp/WEB-INF/jsp/global/global_variables.jsp#L180)):
```js
var genesets = '<%=WebserviceParserUtils.getGenesetIds(request)%>'.trim();
```